atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -605,8 +601,8 @@ private static PersistentLockService createAndRegisterPersistentLockService(
         return Callback.noOp();
     }
 
-    private static boolean isUsingTimeLock(AtlasDbConfig atlasDbConfig, AtlasDbRuntimeConfig runtimeConfig) {
-        return atlasDbConfig.timelock().isPresent() || runtimeConfig.timelockRuntime().isPresent();
+    private static boolean isUsingTimeLock(AtlasDbRuntimeConfig runtimeConfig) {
+        return runtimeConfig.timelockRuntime().serversList().hasAtLeastOneServer();
```

Comment:
@SerialVelocity for SA - I have some concerns about using this as a proxy to determine whether we want to use timelock. 

See #2647 for context - there exists a case where we start up with zero servers because discovery can't find them yet in a k8s environment. `TransactionManagers` doesn't look like it handles this transition properly, since the raw services are only created once.

We could perhaps have a more explicit flag saying whether to use timelock or not - I don't have a very good solution offhand.

Reply:
The presence of the `timelockRuntime` block acted as that flag, before this change, right?